opentelemetry-sdk/src/opentelemetry/sdk/context/propagation/b3_format.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from opentelemetry.context.propagation.httptextformat import HTTPTextFormat
+import opentelemetry.trace as trace
+
+
+class B3Format(HTTPTextFormat):
+    """Propagator for the B3 HTTP header format.
+
+    See: https://github.com/openzipkin/b3-propagation
+    """
+
+    SINGLE_HEADER_KEY = "b3"
+    TRACE_ID_KEY = "x-b3-traceid"
+    SPAN_ID_KEY = "x-b3-spanid"
+    SAMPLED_KEY = "x-b3-sampled"
+    FLAGS_KEY = "x-b3-flags"
+    _SAMPLE_PROPAGATE_VALUES = set(["1", "True", "true", "d"])
+
+    @classmethod
+    def extract(cls, get_from_carrier, carrier):
+        trace_id = trace.INVALID_TRACE_ID
```

Comment:
Note that `INVALID_TRACE_ID` is an `int` with value `0`, but the getter will return a `str`. Personally, I'd use `None` or an empty string here, to have a `str` or `Optional[str]` instead of a `Union[str, int]`. But that's just a suggestion, it should work as-is too.

Reply:
I see the point here. I'm not too familiar with how mypy deals with type changes, but I think it makes sense to enforce a string.

I'll format the ints in the expected format.